docs/ci.md

```diff
@@ -169,25 +169,51 @@ $ subscription=<mySubscription>
 $ subscription_id=<mySubscriptionID>
 $ resourcegroup=<myResourceName>
 $ location=<myLocation>
-
-# Set subscription so that we don't need to specify it at every command
+$ app_name=<myAppName>
+# federated_name should not have spaces!
+$ federated_name=<myFederatedCredentialName>
+$ organization=<myGitHubOrganization>
+$ repository=<myGitHubRepository>
+$ environment=<myCIJobEnvironment>
+
+# Set subscription so that we don't need to specify it at every command.
 $ az account set --subscription $subscription
 
-# Create resource group
+# Create resource group.
+# This is not needed to generate secrets but it is mandatory to creates AKS
+# cluster in the CI.
 $ az group create --name $resourcegroup --location $location
 
-# Create the needed secrets to be able to "az login" from the CI.
-$ az ad sp create-for-rbac --name inspektor-gadget-ci --role contributor \
-	--scopes /subscriptions/${subscription_id}/resourceGroups/${resourcegroup} \
-	--sdk-auth
+# Register an application for your CI.
+$ az ad app create --display-name $app_name
+{
+# It should reply with a big JSON object.
+}
+
+# Let's get the ID of the created application
+$ app_id=$(az ad app list --display-name $app_name --query [0].id | tr -d '"')
+
+# Create the federated credential to be able to "az login" from the CI.
+$ az ad app federated-credential create --id $app_id --parameters <(echo "{
+  \"name\": \"${federated_name}\",
+  \"issuer\": \"https://token.actions.githubusercontent.com\",
+  \"subject\": \"repo:${organization}/${repository}:environment:${environment}\",
+  \"description\": \"AKS federated credentials for CI\",
+  \"audiences\": [
+    \"api://AzureADTokenExchange\"
+  ]
+}")
 {
-# The JSON object containing a secret should be stored as CI secret.
+# It replies with a JSON object which has name set to $federated_name.
 }
 ```
 
 After doing this, you will need to create several secrets:
 
-1. `AZURE_AKS_CREDS`: It stores the JSON object outputted by ``.
+1. `AZURE_AKS_CLIENT_ID`: The application ID as given by `az ad app list --display-name $app_name --query [0].id`.
```

Comment:
```suggestion
1. `AZURE_AKS_CLIENT_ID`: The application ID as given by `az ad app list --display-name $app_name --query [0].appId`.
```

Reply:
> It seems both of them do not work.

This is weird. I used the exact steps in [different project](https://github.com/Azure/kubectl-aks/pull/46) to see if it works and everything is green. I wondering this might be linked to how app is registered. I will see if I find anything!